### PR TITLE
[INLONG-600]Multiplexed connection pool for Go sdk

### PR DESCRIPTION
**Pull request by TszKitLo40**
Signed-off-by: Zijie Lu <wslzj40@gmail.com>
In this PR, I implement a multiplexed connection pool for Go sdk.

### CODE REVIEW

**Comment by TszKitLo40** [comment]
@gosonzhang @charlely PTAL

File: tubemq-client-twins/tubemq-client-go/codec/codec.go [hunk 1]
```diff
@@ -0,0 +1,107 @@
+package codec
```
Comment (gosonzhang) [inline tubemq-client-twins/tubemq-client-go/codec/codec.go:1]: Need to add the LICENSE description of apache v2 to the header part of all files

Comment (TszKitLo40) [reply tubemq-client-twins/tubemq-client-go/codec/codec.go:1]: sorry, I forgot this. I will added the license.

File: tubemq-client-twins/tubemq-client-go/pool/multiplexed.go [hunk 2]
```diff
@@ -0,0 +1,386 @@
...
+	return conn, dialOpts, err
+}
+
+func dialWithTimeout(opts *DialOptions) (net.Conn, error) {
+	if len(opts.CACertFile) == 0 {
+		return net.DialTimeout(opts.Network, opts.Address, opts.Timeout)
+	}
+
+	tlsConf := &tls.Config{}
+	if opts.CACertFile == "none" { // 不需要检验服务证书
```
Comment (gosonzhang) [inline tubemq-client-twins/tubemq-client-go/pool/multiplexed.go:347]: Comments in English are required

Comment (TszKitLo40) [reply tubemq-client-twins/tubemq-client-go/pool/multiplexed.go]: I have removed this comment because it is obvious.

**Review by gosonzhang — APPROVED** [review]
+1

**Comment by gosonzhang** [comment]
@TszKitLo40 I just found out that your modification is merged into the Master. You have a lot of changes, you should merge the branch first, and then merge into the master after the overall debugging is passed. I will create a branch for you INLONG-25 first, please resubmit pr merged into the branch, thank you!

**Pull request closed without merging**